src/opcode.rs

```diff
@@ -1847,3 +1847,159 @@ opcode! {
+        sqe.__bindgen_anon_1.off = len;
+        Entry(sqe)
+    }
+}
+
+// === 6.10 ===
+
+opcode! {
+    /// Send a bundle of messages on a socket in a single request.
+    pub struct SendBundle {
+        fd: { impl sealed::UseFixed },
+        buf_group: { u16 },
+        ;;
+        flags: i32 = 0,
+        len: u32 = 0
+    }
+
+    pub const CODE = sys::IORING_OP_SEND;
+
+    pub fn build(self) -> Entry {
+        let SendBundle { fd, len, flags, buf_group } = self;
+
+        let mut sqe = sqe_zeroed();
+        sqe.opcode = Self::CODE;
+        assign_fd!(sqe.fd = fd);
+        sqe.len = len;
+        sqe.__bindgen_anon_3.msg_flags = flags as _;
+        sqe.ioprio |= sys::IORING_RECVSEND_BUNDLE as u16;
+        sqe.flags |= 1 << sys::IOSQE_BUFFER_SELECT_BIT;
+        sqe.__bindgen_anon_4.buf_group = buf_group;
+        Entry(sqe)
+    }
+}
+
+opcode! {
+    /// Receive a bundle of buffers from a socket.
+    ///
+    /// Parameter
+    ///     buf_group: The id of the provided buffer pool to use for the bundle.
+    ///
+    /// Note that as of kernel 6.10 first recv always gets a single buffer, while second
+    /// obtains the bundle of remaining buffers. This behavior may change in the future.
+    ///
+    /// Bundle variant is available since kernel 6.10
+    pub struct RecvBundle {
+        fd: { impl sealed::UseFixed },
+        buf_group: { u16 },
+        ;;
+        flags: i32 = 0
+    }
+
+    pub const CODE = sys::IORING_OP_RECV;
+
+    pub fn build(self) -> Entry {
+        let RecvBundle { fd, buf_group, flags } = self;
+
+        let mut sqe = sqe_zeroed();
+        sqe.opcode = Self::CODE;
+        assign_fd!(sqe.fd = fd);
+        sqe.__bindgen_anon_3.msg_flags = flags as _;
+        sqe.__bindgen_anon_4.buf_group = buf_group;
+        sqe.flags |= 1 << sys::IOSQE_BUFFER_SELECT_BIT;
+        sqe.ioprio |= sys::IORING_RECVSEND_BUNDLE as u16;
+        Entry(sqe)
+    }
+}
+
+opcode! {
+    /// Receive multiple messages from a socket as a bundle.
+    ///
+    /// Parameter:
+    ///     buf_group: The id of the provided buffer pool to use for each received message.
+    ///
+    /// MSG_WAITALL should not be set in flags.
+    ///
+    /// The multishot version allows the application to issue a single receive request, which
+    /// repeatedly posts a CQE when data is available. Each CQE will take a bundle of buffers
+    /// out of a provided buffer pool for receiving. The application should check the flags of each CQE,
+    /// regardless of its result. If a posted CQE does not have the IORING_CQE_F_MORE flag set then
+    /// the multishot receive will be done and the application should issue a new request.
+    ///
+    /// Note that as of kernel 6.10 first CQE always gets a single buffer, while second
+    /// obtains the bundle of remaining buffers. This behavior may change in the future.
+    ///
+    /// Multishot bundle variant is available since kernel 6.10.
+    pub struct RecvMultiBundle {
```

Comment:
I'm not sure if we want too many opcodes, it looks exactly like RecvMulti just with one more flag

Reply:
I'm not sure what would be a better alternative in the current design. Exposing ioprio in API so that user could set IORING_RECVSEND_BUNDLE is quite messy as well as additional argument to existing calls that will break things.
Also keep in mind that RecvMultiBundle is available starting from kernel 6.10 while RecvMulti appeared back in 6.0.

It feels to me that having opcodes for those use cases is actually cleaner.